core/src/program/blocks/span_block.rs

```diff
@@ -16,25 +16,35 @@ const MAX_OPS_PER_BATCH: usize = GROUP_SIZE * BATCH_SIZE;
 
 // SPAN BLOCK
 // ================================================================================================
-/// A code block used to describe a linear sequence of operations (i.e., no branching or loops).
+/// Block for a linear sequence of operations (i.e., no branching or loops).
 ///
-/// When the VM executes a Span block, it breaks the sequence of operations into batches and
-/// groups according to the following rules:
-/// - A group may contain up to 9 operations or a single immediate value.
-/// - An operation carrying an immediate value cannot be in the 9th position in a group.
-/// - A batch may contain up to 8 groups.
-/// - There is no limit on the number of batches contained within a single span.
+/// Executes its operations in order. Fails if any of the operations fails.
 ///
-/// Thus, for example, executing 8 pushes in a row will result in two operation batches:
-/// - The first batch will contain 8 groups, with the first group containing 7 push opcodes,
-///   and the remaining 7 groups containing immediate values for each of the push operations.
-/// - The second batch will contain 2 groups, with the first group containing a single push opcode,
-///   and the second group containing the immediate value for the last push operation.
+/// A span is composed of operation batches, operation batches are composed of operation groups,
+/// operation groups encode the VM's operations and immediate values. These values are created
+/// according to these rules:
 ///
-/// If a sequence of operations does not have any operations which carry immediate values, then
-/// up to 72 operations can fit into a single batch.
+/// - A span contains one or more batches.
+/// - A batch contains exactly 8 groups.
+/// - A group contains exactly 9 operations or 1 immediate value.
+/// - NOPs are used to fill a group or batch when necessary.
```

Comment:
nit: we usually write `NOOP` with two "O"'s.

Reply:
done :)